src/main.c

```diff
@@ -1,69 +1,123 @@
 #include <stdlib.h>
 #include <stdio.h>
 
-struct node {
-    struct node *pnext;
-    int item;
-};
+/* node data is stored here */
+#include "node.h"
```

Comment:
I didn't end up actually using this at all. Is it better style to use a header in that way? Maybe put the add functions and such there?

Reply:
never put any code in a header file. It should only contain data definitions (like struct defs) and function prototypes, so other files can know about those

Reply:
Ok. What's a function prototype? Analogous to a Java abstract method?

Reply:
no, there are no abstract methods as there is no inheritance or objects or anything of that sort.

A function prototype i just telling you the type of a function, but not giving you the function declaration.

For example, lets say I wanted to call the function `puts`. 
if i just said 
`puts("hi")` in my program, the compiler would complain, because it has no idea that the puts function exists, and assumes that i've made an error.
So, you have to have a function prototype above your use which shows what type it is, e.g.

```
void puts(char *);

int main() {
puts("test\n");
}
```
should compile just fine.

This is what headers do, they contain no code, but they tell your program what it needs to do to interact with the files that the headers describe.
They contain mostly function prototypes and struct definitions, so that your program can call those functions and use those structs, but nothing more (except perhaps macros, but using macros is usually bad practice).